src/raft.c

```diff
@@ -347,6 +348,17 @@ void RaftExecuteCommandArray(RedisRaftCtx *rr,
         size_t cmdlen;
         const char *cmd = RedisModule_StringPtrLen(c->argv[0], &cmdlen);
 
+        if (strncasecmp("watch", cmd, cmdlen) == 0) {
+            unsigned long long id = cmds->client_id;
+
+            int nokey;
+            RedisModule_DictGetC(rr->session_dict, &id, sizeof(id), &nokey);
+            if (nokey) {
+                void *session = RedisModule_Alloc(sizeof(void *));
+                RedisModule_DictSetC(rr->session_dict, &id, sizeof(id), session);
```

Comment:
Not sure if this is the same thing but I think, instead of `session` in libraft, we can call it `client_id`.

```c
struct raft_entry_t { 
    ...
    raft_client_id client_id; // Instead of raft_session session
}
```
I think this is easier to understand and also it matches with the function name of redis: `RM_GetClientId()` -> use the return value to set client id in the entry. 

You set the client id of the entry, then do whatever you want to do with it, implement sessions as we do. 

Now it requires libraft change but if we all agree it is better, we can quickly rename it? Please comment.

Reply:
so the argument i made before is that libraft shouldn't be tied to redis semantics.  in redis a "session" is tied to a client id, but it doesn't have to be that way in other systems.  I'm not tied to this interpretation, but I feel "session" should be generic, while the implementation on top of libraft should refer to it whatever makes the most sense for it.

I can change the session variable here to be more understandable.

Reply:
session -> client_session and functions are *ClientSession()

Reply:
I thought we would change variable name in libraft, no?

To be honest, I don't like session->client_session change. In general I don't like verbose naming as it makes code larger/longer. `session` was perfectly clear to me, "client_session" doesn't make it more readable imho. 

Just to be clear, my point was referring `client_id` <-> `session` <-> `session_id` interchangeably in the code. I find this a bit confusing / harder to understand. I think confusion comes from `raft_entry_t` as it has `session` variable (it is an id) and we also have "session" struct/logic in redisraft. To fix this issue, we can rename `raft_entry_t`'s `session` to `client_id`. Then, it'll be easier to understand, at least this was my confusion / problem.

Anyway, I'm not even sure we discuss same thing :) Maybe I misunderstood initial comment in this thread.

Reply:
We agreed in our daily to change raft session into session id, and then to solve the ambiguity, but we can do it later.